### ISSUE TITLE
[java] UnusedPrivateMethod does not recognize Lombok @EqualsAndHashCode.Include annotation

### ISSUE THREAD

**Issue by mladjan-gadzic** [pmd/pmd#3359]
**Affects PMD Version:** 
6.35.0+
7.0.0

**Rule:**
[UnusedPrivateMethod](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedprivatemethod)

**Description:**
PMD does not recognize `@EqualsAndHashCode.Include` Lombok annotation (https://projectlombok.org/features/EqualsAndHashCode)

**Code Sample demonstrating the issue:**

```java
import lombok.EqualsAndHashCode;

@EqualsAndHashCode
public class Foo {

  @EqualsAndHashCode.Exclude
  private BigDecimal bar;

  @EqualsAndHashCode.Include
  private BigDecimal getBarValueForEqAndHc() { // false positive UnusedMethod
    return bar != null ? bar.stripTrailingZeros() : null;
  }
}
```

**Expected outcome:**

PMD reports a violation at line 8, but that's wrong. That's a false positive.

**Running PMD through:** 
Gradle

**Comment by adangel** [comment]
Note, you can configure the property `ignoredAnnotations` like so:

```xml
<rule ref="category/java/bestpractices.xml/UnusedPrivateMethod">
    <properties>
        <property name="ignoredAnnotations" value="java.lang.Deprecated,jakarta.annotation.PostConstruct,jakarta.annotation.PreDestroy,lombok.EqualsAndHashCode.Include,lombok.EqualsAndHashCode.Exclude" />
    </properties>
</rule>
```

To solve this issue, we need to decide, whether we add these lombok annotations to the default value of the property. Which probably makes sense.
